### ISSUE TITLE
UPDATE events may be missing some columns

### ISSUE THREAD

**Issue by ianschmitz** [openartmarket/supabase-live-table#2]
Hi there :wave:. Thanks for creating this lib. I'm surprised something like this isn't built into Supabase yet!

I discovered an issue with the implementation here. When the updated record is received, the record is overwritten in the `recordById` map. https://github.com/openartmarket/supabase-live-table/blob/34882436d6316e86bbecb895bf3397213bb0cda2/src/index.ts#L186

This creates an issue as not all columns may be sent during the update due to the way Postgres handles large column values. You can see [an existing issue](https://github.com/supabase/realtime/issues/223) over on the Supabase realtime repo. Setting `REPLICA IDENTITY` as suggested i don't think would fix the problem, and potentially introduces performance issues.

May i suggest that we merge the old and new record, such that missing columns don't get lost?

Something like:

```diff
-this.recordById.set(record.id, record);
+const mergedRecord = { ...this.recordById.get(record.id), ...record };
+this.recordById.set(record.id, mergedRecord);
```

For what it's worth i've forked your code and made this change and it's been working great on my end. Not sure if there are cases where it may break?

**Comment by aslakhellesoy** [comment]
Good catch @ianschmitz! Please see #3 for a fix. In particular the [sequence diagram](https://github.com/openartmarket/supabase-live-table/blob/2-merge-on-update/docs/sequence-diagrams/merges-updates-with-previous-value.md) generated from the new test.

If you're happy with this I'll merge and release.

**Comment by aslakhellesoy** [comment]
> Hi there 👋. Thanks for creating this lib. I'm surprised something like this isn't built into Supabase yet!

It would indeed be nice if this library was built into Supabase. OTOH, it might be a bit too specific to warrant an inclusion there.

But if someone from Supabase discovers it and decides to include it, I'd be pumped!

**Comment by aslakhellesoy** [comment]
By the way @ianschmitz - would you be able to share what product you're using this library in? I'd love to list it under the [in the wild](https://github.com/openartmarket/supabase-live-table#in-the-wild) section.

**Comment by aslakhellesoy** [comment]
Fixed by #3 and released in 0.1.0
